pipeline/pipeline_runner.go

```diff
@@ -158,7 +158,7 @@ type PipelinePack struct {
 // Returns a new PipelinePack pointer that will recycle itself onto the
 // provided channel when a message has completed processing.
 func NewPipelinePack(recycleChan chan *PipelinePack) (pack *PipelinePack) {
-	msgBytes := make([]byte, message.MAX_MESSAGE_SIZE)
+	msgBytes := make([]byte, 0, message.MAX_MESSAGE_SIZE)
```

Comment:
"our code generally copies into MsgBytes and then resizes down" Can you point me to where you are referring, The length is meaningless when populating it with something new and we should only be dealing with capacity (like this):

```Go
		messageLen := len(unframed)
		if messageLen > cap(pack.MsgBytes) {
			pack.MsgBytes = make([]byte, messageLen)
		}
		pack.MsgBytes = pack.MsgBytes[:messageLen]
		copy(pack.MsgBytes, unframed)
```
Also if read_message("raw") is called on a new empty message it should not return MAX_MESSAGE_SIZE of NUL bytes (which is the reason the change is included here)